src/katgpucbf/fgpu/kernels/ddc.mako

```diff
@@ -202,18 +199,15 @@ void ddc(
         }
     }
 
-    // Compute the mixer for the first sample output by this workitem
-    mix_bias += get_global_id(0) * (C * SUBSAMPLING) * mix_scale;
-    mix_bias -= rint(mix_bias);
-    float2 mix_base;
-    sincospif(2 * (float) mix_bias, &mix_base.y, &mix_base.x);
-
+    unsigned int mix_cycles = get_global_id(0) * C * mix_scale + mix_bias;
 #pragma unroll
     for (int i = 0; i < C; i++)
     {
-        accum[i] = cmul(accum[i], mix_base);
-        if (i > 0)
-            accum[i] = cmul(accum[i], local.mix_lookup[i]);
+        float2 mix;
+        // Casting from unsigned int to int changes the range from [0, 2pi) to [-pi, pi).
+        __sincosf(2 * (float) M_PI / 4294967296.0f * (int) mix_cycles, &mix.y, &mix.x);
```

Comment:
2*pi to turn cycles into radians. Divide by 2^32 to turn fixed-point into a real number.

Reply:
Ah right, I didn't recognise 2**32 initially. Please proceed, nothing to see here.

Reply:
I've added a comment to make that more obvious, as well as updating another comment. Please take another quick look.

Reply:
If one wanted to be a real pedant, one would say "convert fixed-point to floating point", rather than "real".

But I don't feel too strongly about this.